examples/gno.land/r/demo/boards2/post.gno

```diff
@@ -158,12 +159,22 @@ func (post *Post) AddRepostTo(creator std.Address, title, body string, dst *Boar
 		panic("cannot repost non-thread post")
 	}
 
+	if post.isHidden {
+		panic("thread has been flagged as inappropriate")
+	}
+
 	pid := dst.incGetPostID()
 	repost := newPost(dst, pid, creator, title, body, pid, post.id, post.board.id)
-	dst.threads.Set(pid.Key(), repost)
-	if !dst.IsPrivate() {
-		post.reposts.Set(dst.id.Key(), pid)
-	}
+
+	args := Args{creator, dst.id, pid}
+	dst.perms.WithPermission(creator, PermissionThreadRepost, args, func(_ Args) {
+		post.repostsCount++
+		dst.threads.Set(pid.Key(), repost)
+		if !dst.IsPrivate() {
+			post.reposts.Set(dst.id.Key(), pid)
+		}
+	})
```

Comment:
What do you think about moving the `WithPermission` call to the public `CreateRepost` function? It would simplify the `Post` type logic, be at the same time consistent with the other public functions and would also make the `WithPermission` call more explicit.

Apart from that we have to consider that the public `CreateRepost` function returns an ID. The repost with the returned ID might not exist when the call finishes if the permissions implementation uses for example a proposal to create the repost, but that's fine for now, it was discussed with Jae in one of the initial calls. @x1unix you can check `CreateBoard` as example.

Reply:
Ok